KeenClient/KeenClient.m

```diff
@@ -335,34 +295,36 @@ - (id)initWithProjectID:(NSString*)projectID
 + (KeenClient *)sharedClientWithProjectID:(NSString *)projectID
                               andWriteKey:(NSString *)writeKey
                                andReadKey:(NSString *)readKey {
-
     // Validate key parameters
-    if (![KeenClient validateProjectID:projectID]) {
-        KCLogError(@"Invalid projectID: %@", projectID);
+    if (projectID == nil || projectID.length <= 0) {
+        KCLogError(@"You must provide a projectID.");
         return nil;
     }
 
-    if (nil != writeKey && // only validate a non-nil value
-        ![KeenClient validateKey:writeKey]) {
-        KCLogError(@"Invalid writeKey: %@", writeKey);
+    if (writeKey && writeKey.length <= 0) {
+        KCLogError(@"Your writeKey cannot be an empty string.");
         return nil;
     }
 
-    if (nil != readKey && // only validate a non-nil value
-        ![KeenClient validateKey:readKey]) {
-        KCLogError(@"Invalid readKey: %@", readKey);
+    if (readKey && readKey.length <= 0) {
+        KCLogError(@"Your readKey cannot be an empty string.");
         return nil;
     }
 
+    // If the keys are already set, return the existing instance, don't overwrite the keys
+    if (self.sharedClient.projectID || self.sharedClient.writeKey || self.sharedClient.readKey) {
```

Comment:
This would technically require revving a major revision, maybe this part of the change could be pared off and thrown into a `vnext_major` branch until we're ready to release a major revision then bring it to master?